haystack/core/pipeline/pipeline.py

```diff
@@ -21,8 +27,9 @@ class Pipeline(PipelineBase):
     Orchestrates component execution according to the execution graph, one after the other.
     """
 
+    @staticmethod
```

Comment:
This is a kinda of a scary change 😨

Reply:
Not that scary, just a refactor to make the function take things in explicitly rather than through context.